From a857d0a5a9eaa6cb4ad511ebb47fa3f2286466a9 Mon Sep 17 00:00:00 2001
From: Amir Vadai <avadai@redhat.com>
Date: Thu, 11 Sep 2014 11:59:55 -0400
Subject: [ethernet] mlx4: Fix selftest failing on non 10G link speed

Message-id: <aec04395f1f145972ac72f6edc446f53ad569895.1410425016.git.avadai@redhat.com>
Patchwork-id: 93708
O-Subject: [Patch RHEL7.1 V1 059/197] net/mlx4_en: Fix selftest failing on non 10G link speed
Bugzilla: 1107617 1107618 1107619 1107620
RH-Acked-by: Jes Sorensen <Jes.Sorensen@redhat.com>

From: Amir Vadai <amirv@mellanox.com>

Connect-X devices selftest speed test shouldn't fail on 1G and 40G link
speeds.

Signed-off-by: Amir Vadai <amirv@mellanox.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
(cherry picked from commit 313c2d375b1c9b648d9d4b96ec1b8185ac6a78c5)
Signed-off-by: Amir Vadai <avadai@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/mellanox/mlx4/en_selftest.c b/drivers/net/ethernet/mellanox/mlx4/en_selftest.c
index c11d063..03e5f6a 100644
--- a/drivers/net/ethernet/mellanox/mlx4/en_selftest.c
+++ b/drivers/net/ethernet/mellanox/mlx4/en_selftest.c
@@ -129,8 +129,10 @@ static int mlx4_en_test_speed(struct mlx4_en_priv *priv)
  if (mlx4_en_QUERY_PORT(priv->mdev, priv->port))
   return -ENOMEM;
 
- /* The device currently only supports 10G speed */
- if (priv->port_state.link_speed != SPEED_10000)
+ /* The device supports 1G, 10G and 40G speeds */
+ if (priv->port_state.link_speed != 1000 &&
+     priv->port_state.link_speed != 10000 &&
+     priv->port_state.link_speed != 40000)
   return priv->port_state.link_speed;
  return 0;
 }
-- 
1.7.1